constraints/src/main/java/gov/nasa/jpl/aerie/constraints/json/ConstraintParsers.java

```diff
@@ -274,6 +275,31 @@ static JsonParser<Transition> transitionP(JsonParser<ProfileExpression<?>> profi
               microseconds -> Duration.of(microseconds, Duration.MICROSECONDS),
               duration -> duration.in(Duration.MICROSECONDS));
 
+  public static final JsonParser<Interval> intervalP =
+      productP
+          .field("start", durationP)
+          .field("end", durationP)
+          .field("startInclusivity", inclusivityP)
+          .field("endInclusivity", inclusivityP)
+          .map(
+              untuple((start, end, startInclusivity, endInclusivity) -> Interval.between(start, startInclusivity, end, endInclusivity)),
+              $ -> tuple($.start, $.end, $.startInclusivity, $.endInclusivity)
+          );
+
+  public static final JsonParser<Violation> violationP =
```

Comment:
Minor: Why are these both `public` rather than `package-private` like the other parsers?

Reply:
I don't remember why `intervalP` was switched, but I changed it back and it seems fine.

`violationP` is public now since we're using it outside of the `constraints` module.